StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/WalletButtonsView/WalletButtonsView.swift

```diff
@@ -12,40 +12,57 @@ import UIKit
 @_spi(STP) public struct WalletButtonsView: View {
     let flowController: PaymentSheet.FlowController
     let confirmHandler: (PaymentSheetResult) -> Void
-    @State private var showingApplePay: Bool
-    @State private var showingLink: Bool
+    let orderedWallets: [ExpressType]
 
-    init(showingApplePay: Bool = false,
-         showingLink: Bool = false,
-         flowController: PaymentSheet.FlowController,
-         confirmHandler: @escaping (PaymentSheetResult) -> Void) {
+    @_spi(STP) public init(flowController: PaymentSheet.FlowController,
+                           confirmHandler: @escaping (PaymentSheetResult) -> Void) {
         self.confirmHandler = confirmHandler
         self.flowController = flowController
-        _showingApplePay = State(initialValue: showingApplePay)
-        _showingLink = State(initialValue: showingLink)
+
+        // Determine available wallets and their order from elementsSession
+        var wallets: [ExpressType] = []
+        for type in flowController.elementsSession.orderedPaymentMethodTypesAndWallets {
+            switch type {
+            case "link":
+                // Also check PaymentSheet local availability logic
+                if PaymentSheet.isLinkEnabled(elementsSession: flowController.elementsSession, configuration: flowController.configuration) {
+                    wallets.append(.link)
+                }
+            case "apple_pay":
+                if PaymentSheet.isApplePayEnabled(elementsSession: flowController.elementsSession, configuration: flowController.configuration) {
+                    wallets.append(.applePay)
+                }
+            default:
+                break
```

Comment:
If we introduced a new wallet type, seems like we would prefer to `continue` instead -- otherwise, it's possible to forget appending `.applepay` and/or `.link`

Reply:
I think the behavior is the same either way (it's in the `switch` not the `for` loop) but I can swap it out if it's less confusing